libbeat/processors/add_host_metadata/add_host_metadata.go

```diff
@@ -71,10 +84,68 @@ func (p *addHostMetadata) loadData() {
 		if p.info.OS.Build != "" {
 			p.data.Put("host.os.build", p.info.OS.Build)
 		}
+
+		if p.config.NetInfoEnabled {
+			// IP-address and MAC-address
+			var ipList, hwList, err = p.getNetInfo()
+			if err != nil {
+				logp.Warn("Error when getting network information %v", err)
+			}
+
+			if len(ipList) > 0 {
+				p.data.Put("host.ip", ipList)
+			}
+			if len(hwList) > 0 {
+				p.data.Put("host.mac", hwList)
+			}
+		}
+
 		p.lastUpdate = time.Now()
 	}
 }
 
+func (p addHostMetadata) getNetInfo() ([]string, []string, error) {
+	var ipList []string
+	var hwList []string
+
+	// Get all interfaces and loop through them
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		return ipList, hwList, err
+	}
+	for _, i := range ifaces {
+		// Skip loopback interfaces
+		if i.Flags&net.FlagLoopback == net.FlagLoopback {
+			continue
+		}
+
+		hw := i.HardwareAddr.String()
+		// Skip empty hardware addresses
+		if hw != "" {
+			hwList = append(hwList, hw)
+		}
+
+		addrs, err := i.Addrs()
+		if err != nil {
+			// If we get an error, log it and continue with the next interface
+			logp.Warn("Error when getting IP address %v", err)
```

Comment:
Lets not log each error but add it to the list. We can use `multierror` here. Outside the loop you can specify `var errs multierror.Errors` and then use `errs = append(errs, err)`. On line 145 you can then return `errs.Err()`.